### PR TITLE
Add user PIN authentication 🔒 

### LINKED ISSUE #11: User authentication
**Description**:

As a user,
I might want to secure my app access with an authentication method or just skip the process
So that I can avoid foreign access to my funds

**Note**:

Using react-native key-value secure store for PIN storage should be sufficient to achieve this.
[https://github.com/cashubtc/eNuts/blob/main/src/storage/store/SecureStore.ts](https://github.com/cashubtc/eNuts/blob/main/src/storage/store/SecureStore.ts)

**Requirements**:

- [x]  PIN setup/skip on initial start
- [x]  PIN confirm after setup on initial start
- [x]  PIN query (if setup by user previously) on initial start
- [x]  Lock screen after 3 failing attemps. Time locked should raise on each failing attempt. (2m, 5m, 10m)
- [x]  Edit/remove/setup PIN in settings/security
- [x] Ask for PIN if app comes to the foreground after beeing > 5 mins in the background.
- [x] Translate the screen

**Enhancement/Ideas**:

1. Auth method selection on initial start (PIN, TouchID, FaceID)
2. Activate/deactive Auth query for a LN TX in settings.

[https://docs.expo.dev/versions/latest/sdk/local-authentication/](https://docs.expo.dev/versions/latest/sdk/local-authentication/)

### PR DESCRIPTION
**Pull request by KKA11010**
Fixes #11

### CODE REVIEW

**Comment by BilligsterUser** [comment]
is this branch ready?

**Comment by KKA11010** [comment]
> 

let me just resolve conflicts quick

**Review by BilligsterUser — CHANGES_REQUESTED** [review]

File: src/components/App.tsx [hunk 1]
```diff
@@ -2,21 +2,25 @@ import { getEncodedToken } from '@cashu/cashu-ts'
 import Button from '@comps/Button'
 import usePrompt from '@comps/hooks/Prompt'
 import { env } from '@consts'
+import { FiveMins } from '@consts/time'
 import { addAllMintIds, getBalance, getContacts, getMintsBalances, getMintsUrls, getPreferences, initDb, setPreferences } from '@db'
 import { fsInfo } from '@db/fs'
 import { l } from '@log'
 import MyModal from '@modal'
 import type { IInitialProps, IPreferences, ITokenInfo } from '@model'
```
Comment (BilligsterUser) [inline src/components/App.tsx:10]: 'IInitialProps' is defined but never used

File: src/context/Pin.tsx [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import { l } from '@log'
+import { createContext, useEffect, useState } from 'react'
```
Comment (BilligsterUser) [inline src/context/Pin.tsx:2]: 'useEffect' is defined but never used

**Review by BilligsterUser — CHANGES_REQUESTED** [review]

File: src/context/Pin.tsx (hunk 2 above)
Comment (KKA11010) [reply src/context/Pin.tsx:2]: fixed

File: src/components/App.tsx (hunk 1 above)
Comment (KKA11010) [reply src/components/App.tsx:10]: fixed

File: src/components/App.tsx [hunk 3]
```diff
@@ -55,6 +66,44 @@ export default function App(){
...
+			const secsPassed = now - lockData.timestamp
+			const lockedTime = lockData.lockedTime - secsPassed
+			setAttempts({
+				...lockData,
+				mismatch: false,
+				lockedTime
+			})
+		}
+		// handle app was longer than 5 mins in the background
+		const bgTimestamp = await store.get('authBg')
```
Comment (BilligsterUser) [inline src/components/App.tsx:99]: pls prefix all store keys 
like you do for settings 
await store.set('settings:lang', code)

**Review by BilligsterUser — CHANGES_REQUESTED** [review]

**Comment by BilligsterUser** [comment]
i pushed a commit to use the store instead of asyncStore

File: src/components/App.tsx (hunk 3 above)
Comment (KKA11010) [reply src/components/App.tsx]: done

File: src/components/App.tsx [hunk 4]
```diff
@@ -207,6 +255,16 @@ function _App() {
 				l('Error while initializing contacts from DB')
 			}
 		}
+		async function initAuth() {
+			const skipped = await store.get('auth:skipped')
+			const pinHash = await secureStore.get('auth:pin')
+			setAuth({
+				shouldAuth: isNull(pinHash) ? '' : pinHash,
```
Comment (BilligsterUser) [inline src/components/App.tsx:262]: i think authHash or pinHash is a better name 
shouldAuth sounds like a bool

**Review by BilligsterUser — CHANGES_REQUESTED** [review]

**Review by BilligsterUser — DISMISSED** [review]

**Review by BilligsterUser — APPROVED** [review]

**Pull request merged**